packages/react-html/README.md

```diff
@@ -12,112 +12,165 @@ $ yarn add @shopify/react-html
+The main entrypoint also exposes components for easily building links to your assets:
 
-```ts
-import {Style, Script} from '@shopify/react-html';
-```
+##### `<Style />`
 
-### Style
+The `<Style />` component lets you render `<link>` tags in your document dynamically as part of your react app. It supports all of the props of a basic `link` tag, but forces some properties to be the values needed for a stylesheet.
 
-The `<Style />` component lets you render `<link>` tags in your document dynamically as part of your react app.
+```tsx
+import {Style} from '@shopify/react-html';
 
-```ts
 <Style
   href="./some-style.css"
   integrity="some-integrity-hash"
   crossOrigin="anonymous"
-/>
+/>;
 ```
 
-### Script
+##### `<Script />`
+
+The `<Script />` component lets you render `<script>` tags in your document dynamically as part of your react app. It supports all the props of a basic `script` tag.
 
-The `<Script />` component lets you render `<script>` tags in your document dynamically as part of your react app.
+```tsx
+import {Script} from '@shopify/react-html';
 
-```ts
 <Script
   src="./some-script.js"
   integrity="some-integrity-hash"
   crossOrigin="anonymous"
-/>
+/>;
 ```
+
+### Server
+
+The server entrypoint allows you to easily construct an HTML document for the response. It also re-exports all of the components noted above.
+
+#### `<Html />` and `render()`
+
+The `<Html>` component serves as a top level wrapper for a React application, allowing you to avoid needing any kind of server-side template. The `render()` function creates a stringified version of the HTML document with an appropriate DOCTYPE.
+
+```tsx
+import * as React from 'react';
+import {render, Html} from '@shopify/react-html/server';
+
+import App from '../app';
+
+export default function middleware(ctx, next) {
+  ctx.body = render(
+    <Html>
+      <App />
+    </Html>,
+  );
+
+  await next();
+}
+```
+
+The component will automatically propagate any usage of the [`react-helmet` library](https://github.com/nfl/react-helmet) in your app’s content to manipulate the title or other top level HTML or HEAD attributes.
+
+##### Props
+
+The `<Html />` component accepts the following props:
+
+- `manager`: a `Manager` instance. When provided, the `Html` component will extract all the information from this object and place it in an appropriate place in the document.
+- `children`: the application. It will be rendered to a string and placed inside a `div` with an ID of `app`.
+- `locale`: the language to use for the HTML `lang` attribute.
+- `styles`: descriptors for any style tags you want to include in the HEAD of the document.
+- `scripts`: descriptors for any script tags you want to include in your document. All scripts passed to this property will be deferred by appending them to the end of the document. We encourage this as a default, although you may use `blockingScripts` for any scripts that must be included in the HEAD of the document.
```

Comment:
I'd end this bullet at `We encourage this as a default`(and maybe link to an article or something on why), since `blockingScripts` is the next item in the list anyway.